File: ts_sdk/src/resource_management/fs_records/claude/claude-settings/github-repos.ts
/**
 * ClaudeGithubReposFsRecord — GitHub repos metadata from claude settings.
 */
import { FsRecord, type FsRecordData } from '../../fs-record';
import { RecordType } from '../../record-types';
import { StorageLayout } from '../../storage-layout';
import { fsRecordTypeRegistry } from '../../record-type-registry';

export class ClaudeGithubReposFsRecord extends FsRecord {
  static override _recordType = RecordType.CLAUDE_SETTINGS_GITHUB_REPOS;
  static override _readOnly = true;
  static override _storageLayout = StorageLayout.LIST_ITEM;

  repos?: Record<string, unknown>[];
  last_fetched?: string;

  constructor(data?: Partial<FsRecordData>) {
    super(data);
  }
}

fsRecordTypeRegistry.register(ClaudeGithubReposFsRecord._recordType, ClaudeGithubReposFsRecord as any);

File: flow_sdk/fs_records/claude/claude_settings/github_repos.py
"""ClaudeGithubReposFsRecord -- GitHub repo path mappings from ~/.claude.json."""

from __future__ import annotations

from typing import Any, ClassVar

from flow_sdk.fs_store import Record, RecordType


class ClaudeGithubReposFsRecord(Record):
    """Maps GitHub repo slugs to local filesystem paths."""

    _record_type: ClassVar[str] = RecordType.CLAUDE_SETTINGS_GITHUB_REPOS

    def __init__(self, **kwargs: Any):
        if "type" not in kwargs:
            kwargs["type"] = RecordType.CLAUDE_SETTINGS_GITHUB_REPOS
        super().__init__(**kwargs)
        from flow_sdk.fs_store.fs_ref import FSRef
        object.__setattr__(self, "_asset_ref", FSRef("/", read_only=True))

    @property
    def repos(self) -> dict[str, list[str]]:
        return object.__getattribute__(self, "__dict__").get("repos") or {}

    @property
    def repo_count(self) -> int:
        """Number of distinct GitHub repos tracked."""
        return len(self.repos)

    def get_paths(self, repo: str) -> list[str]:
        """Return local paths for a given repo slug, or empty list."""
        return self.repos.get(repo, [])

    @classmethod
    def from_raw(cls, data: dict) -> ClaudeGithubReposFsRecord:
        """Create from the githubRepoPaths sub-object."""
        rec = cls(repos=dict(data))
        rec.id = "default"
        return rec
